### ISSUE TITLE
backend/remote: Unable to taint remote resources in local execution mode

### ISSUE THREAD

**Issue by dekimsey** [hashicorp/terraform#27480]
### Terraform Version

```
Terraform v0.14.4
```

### Terraform Configuration Files

```terraform
terraform {
  required_version = ">= 0.13"

  backend "remote" {
    organization = "trustwave"

    workspaces {
      prefix = "mystuff-"
    }
  }
}
```

### Debug Output
n/a

### Crash Output
n/a

### Expected Behavior
Taint succeeds

```
$ terraform taint 'module.hot.module.this.aws_instance.this[0]'
Resource instance module.hot.module.this.aws_instance.this[0] has been marked as tainted.
```

### Actual Behavior

Attempts to taint a resource fail.
```
$ terraform taint 'module.hot.module.this.aws_instance.this[0]'
Failed to load state: Remote workspace Terraform version "0.14.2" does not match local Terraform version "0.14.4"
```

### Steps to Reproduce

1. Using terraform remote with "Local" execution
1. `terraform taint my-resource`
2. Set execution to Remote
3. Change Terraform version to the matching version
4. Set execution to Local
5. `terraform taint`

### Additional Context

None

### References

- #27319

**Comment by alisdair** [comment]
Thanks for reporting this, @dekimsey, I'm able to confirm the bug. I'm confused about how it is happening, as it should have been fixed with #27407, but there's something to be fixed here.

In the interim, you can use the `-ignore-remote-version` flag to disable this check.

Note that the reproduction steps must include setting the remote workspace version to an older release. Below are my reproduction steps.

1. Create a configuration:
    ```hcl
    terraform {
      backend "remote" {
        organization = "alisdair-v2"
        workspaces {
          name = "27480"
        }
      }
    }

    resource "null_resource" "none" {
      triggers = {
        foo = uuid()
      }
    }
    ```
2. Run `terraform init` to create the workspace
3. Configure the workspace using the Terraform Cloud UI to use Terraform 0.14.2
4. Run `terraform apply -auto-approve` to create a state file
5. Run `terraform taint null_resource.none`: this fails with an error, but I believe it should not (as 0.14.2 and 0.14.4 are compatible)
6. Set the workspace to local operations using Terraform Cloud UI
7. Run `terraform taint null_resource.none`: this fails with an error, but should not (as the workspace is in local operations mode)

**Comment by dekimsey** [comment]
Thank you for confirming the bug, my apologies on not mentioning the older version that was set an my remote!

I was a bit confused about the fact that I had to swap execution modes to make that change. It wasn't clear in the interface that even with execution mode local the configured terraform version mattered.

**Comment by alisdair** [comment]
Thanks again for reporting this, I really appreciate it. There's a fix for the issue in #27498, and it also affects other commands like `state mv` and `import`, so fixing this is very valuable. If the fix is approved it will go out in the next Terraform 0.14 patch release.
